packages/bot/src/dashboard/controllers.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  getRepoViolationPercentages,
+  getViolatedCommitsPerRepo,
+  Commit,
+} from "../db/models/Commit";
+
+const recentCommitsCount = 10;
+
+interface RepositoryViolationSummary {
+  violationPercentage: string;
+  name: string;
+  commits: Commit[];
+}
+
+type RepositorySummary = RepositoryViolationSummary[];
+
+export async function getRepositorySummary(): Promise<RepositorySummary> {
+  const repoMap: Record<string, RepositoryViolationSummary> = {};
+  const violationPercentages = await getRepoViolationPercentages();
+  for (const { repoName, percentage } of violationPercentages) {
+    repoMap[repoName] = {
+      name: repoName,
+      violationPercentage: percentage + "%",
```

Comment:
The presentation of the percentage (e.g. significant figures, '%' suffix) should be the concern of view (i.e. the template).